Add “representor-device” field to device-info map for PCI and VDPA devs

ykulazhenkov opened the pull request:
We are interested to add new field for PCI devices - “representor-device”. 
This optional field should contain name of the representor device for PCI VF.

The main use-case for now is to report `representor-device` field from [accelerated-brdige-cni](https://github.com/k8snetworkplumbingwg/accelerated-bridge-cni/pull/53) and reflect this info in `k8s.v1.cni.cncf.io/network-status` annotation.

To achieve this `PciDevice` struct in [network-attachment-definition-client](https://github.com/k8snetworkplumbingwg/network-attachment-definition-client/blob/6ace5615d26a4d00d0a24bbbfa5d5f400ce7593e/pkg/apis/k8s.cni.cncf.io/v1/types.go#L60) require update and `network-attachment-definition-client` dependency should be updated in Multus

ykulazhenkov committed:
Add “representor-device” field to device-info for PCI and VDPA devs

This optional key’s value (type string) shall contain the name
of the representor device for PCI VF (e.g. eth3) if available
and applicable.

ykulazhenkov commented:
cc @moshe010 @adrianchiris

adrianchiris commented on SPEC.md:
```diff
@@ -106,6 +106,11 @@ This <ins>optional</ins> key’s value (type string) shall contain the PCI addre
 of the associated physical interface (in standard BDF format “dddd:BB:DD.f”) if
 available and applicable.
 
+#### 3.1.3.5 “vf-representor-device”
```
shall we go with: "device-representor" or "representor-device" ?
im thinking that down the road we would want to reuse this field name for subfunction[1] devices as well as VF.
(perhaps in a different section/type not sure yet, however id prefer it to have the same name)

WDYT ?

work to support this in sriov network device plugin is already in motion [2]

[1] https://www.kernel.org/doc/html/latest/networking/devlink/devlink-port.html#subfunction
[2] https://github.com/k8snetworkplumbingwg/sriov-network-device-plugin/issues/430

ykulazhenkov replied:
Sounds good. I update PR to use `representor-device` field name. We already have `rdma-device` field, so I decided to use `representor-device` instead of `device-representor`.

Also, I pushed commit which updates spec version

ykulazhenkov committed:
Update spec version to "1.1.0"

Changes:
- Add “representor-device” field to device-info map for PCI devs
- Add “representor-device” field to device-info map for VDPA devs

adrianchiris reviewed (approved):
thx for addressing my comment. looks good imo.

lmilleri commented:
It looks good to me

ykulazhenkov commented:
@dougbtv @Billy99 @amorenoz Can we merge this patch if the community doesn't mind?

amorenoz commented on SPEC.md:
```diff
@@ -106,6 +106,11 @@ This <ins>optional</ins> key’s value (type string) shall contain the PCI addre
+This <ins>optional</ins> key’s value (type string) shall contain the name
+of the representor device for PCI VF (e.g. "eth3") if available and applicable.
+
 ### 3.1.4 “vdpa”
```
vdpa devices can also have representor devices. Can you please add the key to it's section as well?

ykulazhenkov replied:
done

amorenoz commented:
Thanks. LGTM

adrianchiris commented:
@amorenoz @dougbtv  can we merge this one ?

amorenoz commented:
I think so, yes.
